cpp/tests/prims/mg_reduce_v.cu

```diff
@@ -0,0 +1,220 @@
+    while (comm_size % row_comm_size != 0) {
+      --row_comm_size;
+    }
+    cugraph::partition_2d::subcomm_factory_t<cugraph::partition_2d::key_naming_t, vertex_t>
+      subcomm_factory(handle, row_comm_size);
+
+    // 2. create MG graph
+
+    if (PERF) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      hr_clock.start();
+    }
+    auto [mg_graph, d_mg_renumber_map_labels] =
+      input_usecase.template construct_graph<vertex_t, edge_t, weight_t, store_transposed, true>(
+        handle, true, true);
+
+    if (PERF) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      double elapsed_time{0.0};
+      hr_clock.stop(&elapsed_time);
+      std::cout << "MG construct_graph took " << elapsed_time * 1e-6 << " s.\n";
+    }
+
+    auto mg_graph_view = mg_graph.view();
+
+    const int hash_bin_count = 5;
+
+    // 3. run MG count if
+
+    if (PERF) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      hr_clock.start();
+    }
+
+    vertex_t const* data = (*d_mg_renumber_map_labels).data();
+    rmm::device_uvector<result_t> test_property(d_mg_renumber_map_labels->size(),
+                                                handle.get_stream());
+    thrust::transform(rmm::exec_policy(handle.get_stream())->on(handle.get_stream()),
+                      data,
+                      data + test_property.size(),
+                      test_property.begin(),
+                      property_transform<vertex_t, result_t>(hash_bin_count));
+    auto vertex_count =
+      reduce_v(handle, mg_graph_view, test_property.begin(), test_property.end(), result_t{0});
+
+    if (PERF) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      double elapsed_time{0.0};
+      hr_clock.stop(&elapsed_time);
+      std::cout << "MG count if took " << elapsed_time * 1e-6 << " s.\n";
+    }
+
+    // 4. compare SG & MG results
+
+    if (prims_usecase.check_correctness) {
+      cugraph::experimental::graph_t<vertex_t, edge_t, weight_t, store_transposed, false> sg_graph(
+        handle);
+      std::tie(sg_graph, std::ignore) =
+        input_usecase.template construct_graph<vertex_t, edge_t, weight_t, store_transposed, false>(
+          handle, true, false);
+      auto sg_graph_view         = sg_graph.view();
+      auto expected_vertex_count = thrust::transform_reduce(
+        rmm::exec_policy(handle.get_stream())->on(handle.get_stream()),
+        thrust::make_counting_iterator(sg_graph_view.get_local_vertex_first()),
+        thrust::make_counting_iterator(sg_graph_view.get_local_vertex_last()),
+        property_transform<vertex_t, result_t>(hash_bin_count),
+        result_t{0},
```

Comment:
Umm... yeah... and we may test with non-zero initial value to check this is properly working.